api/apps/api/src/modules/access-control/projects-acl/project-acl.service.ts

```diff
@@ -92,4 +94,117 @@ export class ProjectAclService implements ProjectAccessControl {
       this.canViewSolutionRoles,
     );
   }
+
+  async checkUserIsOwner(userId: string, projectId: string): Promise<Permit> {
+    const userIsProjectOwner = await this.roles.findOne({
+      where: {
+        projectId,
+        userId,
+        roleName: Roles.project_owner,
+      },
+    });
+    if (!userIsProjectOwner) {
+      return false;
+    }
+
+    return true;
+  }
+
+  async checkLastOwner(userId: string, projectId: string): Promise<Permit> {
+    const allOwnersInProject = await this.roles.find({
+      where: {
+        projectId,
+        roleName: Roles.project_owner,
+      },
+    });
+    if (
+      allOwnersInProject.length === 1 &&
+      allOwnersInProject[0].userId === userId
+    ) {
+      return false;
+    }
+    return true;
+  }
+
+  async findUsersInProject(
+    projectId: string,
+    userId: string,
+  ): Promise<UsersProjectsApiEntity[] | Permit> {
+    const isOwner = await this.checkUserIsOwner(userId, projectId);
+    if (!isOwner) {
+      return isOwner;
+    }
+
+    const usersInProject = await this.roles
+      .createQueryBuilder('users_projects')
+      .leftJoinAndSelect('users_projects.user', 'userId')
+      .where({ projectId })
+      .select([
+        'users_projects.roleName',
+        'userId.displayName',
+        'userId.id',
+        'userId.avatarDataUrl',
+      ])
+      .getMany();
+
+    return usersInProject;
+  }
+
+  async updateUserInProject(
+    projectId: string,
+    updateUserInProjectDto: UserRoleInProjectDto,
+    loggedUserId: string,
+  ): Promise<void | Permit> {
+    const { userId, roleName } = updateUserInProjectDto;
+    const isOwner = await this.checkUserIsOwner(loggedUserId, projectId);
+    if (!isOwner) {
+      return isOwner;
+    }
+
+    const apiDbConnection = getConnection(DbConnections.default);
+    const apiQueryRunner = apiDbConnection.createQueryRunner();
+
+    await apiQueryRunner.connect();
+    await apiQueryRunner.startTransaction();
+
+    try {
+      const existingUserInProject = await this.roles.findOne({
+        where: {
+          projectId,
+          userId,
+        },
+      });
+
+      if (!existingUserInProject) {
+        await this.roles.save({
+          projectId,
+          userId,
+          roleName,
+        });
+      } else {
+        await this.roles.update({ projectId, userId }, { roleName });
+      }
+    } catch (err) {
+      await apiQueryRunner.rollbackTransaction();
+    } finally {
+      await apiQueryRunner.release();
+    }
+  }
+
+  async revokeAccess(
+    projectId: string,
+    userId: string,
+    loggedUserId: string,
+  ): Promise<void | Permit> {
+    const isOwner = await this.checkUserIsOwner(loggedUserId, projectId);
```

Comment:
same suggestion as above here

Reply:
actually if we are at minor styling:

`is....` usually indicates a function (`isBoolean`)

`const owner = await this.isOwner(...)` (i.e. `checkUser` is also redundant)

as of https://github.com/Vizzuality/marxan-cloud/pull/640#discussion_r757312393 and other suggestions, I would actually recommend leaving direct assignment.